File: day1/part2.py
password = 0
startPos = 50
with open("cy.txt") as input:
    position = startPos
    for line in input:
        direction = line[0]
        number = int(line[1:])

        if direction == "R":
            position = position + number
            
            if position >= 100:
                password += position // 100
        else:
            prev = position
            position = position - number
            
            if position <= 0:
                password += position // -100 + 1
                
                if prev == 0:
                    password -= 1
        
        position = position % 100

print(password)

File: day3/part2.js
const fs = require("fs");
const readline = require("readline");

async function parseBanks() {
	const fileStream = fs.createReadStream("./banks.txt");

	const rl = readline.createInterface({
		input: fileStream,
		crlfDelay: Infinity
	});

	let banks = [];
	let totalJoltage = 0;

	for await (const line of rl) {
		let bank = [...line.match(/\d/gm)];
		totalJoltage += joltage(bank);
	}

	console.log(totalJoltage);
}

function joltage(bank) {
	const k = 12;
	const result = [];

	let start = 0;
	let remaining = bank.length;

	for (let needed = k; needed > 0; needed--) {
		let end = remaining - needed;

		let bestDigit = -1;
		let bestIndex = start;

		for (let i = start; i <= end; i++) {
			const d = bank[i] - 0;
			if (d > bestDigit) {
				bestDigit = d;
				bestIndex = i;
			}
		}

		result.push(bestDigit);
		start = bestIndex + 1;
	}

	return Number(result.join(""));
}

parseBanks();